### PR TITLE
Feature/app design system and basic components

### PR DESCRIPTION
**Pull request by joshmarom**

### CODE REVIEW

**Comment by rodolphebertozzo** [comment]
What is exactly?

**Review by matipojo — CHANGES_REQUESTED** [review]
Partial review (42/97)

File: core/app/assets/styles/mixins/_text-truncate.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Utilities: Text truncate
```
Comment (kobizz) [inline core/app/assets/styles/mixins/_text-truncate.scss:1]: What is it used for?

File: core/app/assets/styles/themes/dark/tokens/_dark-theme.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+$eps-dark-gray-300: $color-gray-silver-filigree;
+$eps-dark-gray-200: $color-gray-brainstem;
+$eps-dark-gray-100: $color-gray-yin-bai-silver;
+
+// Theme semantics
+// - data context
+$eps-dark-theme-text-muted: $eps-gray-300;
+$eps-dark-theme-light: $color-black;
+$eps-dark-theme-dark: $color-white;
+$eps-dark-theme-info: null;
```
Comment (kobizz) [inline core/app/assets/styles/themes/dark/tokens/_dark-theme.scss:17]: Why do you use null variables?

File: core/app/assets/styles/tokens/_font.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Font
+//$eps-font-weight-thin: 200;
+//$eps-font-weight-light: 300;
```
Comment (kobizz) [inline core/app/assets/styles/tokens/_font.scss:3]: Don't keep commented-out code. Decide whether to include them or not

File: core/app/assets/styles/tokens/_ratio.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+$embed-responsive-aspect-ratios: () !default;
+// stylelint-disable-next-line scss/dollar-variable-default
+$embed-responsive-aspect-ratios: join(
+						(
+										(21 9),
+										(16 9),
+										(4 3),
+										(1 1),
+						),
```
Comment (kobizz) [inline core/app/assets/styles/tokens/_ratio.scss:13]: Mind the indentation

File: core/app/assets/styles/tokens/_utilities.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Tokens: Utilities
+$displays: none, inline, inline-block, block, table, table-row, table-cell, flex, inline-flex !default;
+$overflows: auto, hidden !default;
+$positions: static, relative, absolute, fixed, sticky !default;
+$user-selects: all, auto, none !default;
```
Comment (kobizz) [inline core/app/assets/styles/tokens/_utilities.scss:5]: Why do you need to re-map all these properties?

File: core/app/assets/styles/tokens/maps/_colors-map.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+				debian:$color-red-debian,
+				carnelian: $color-red-carnelian
+		),
+		yellow:(
+				lightning: $color-yellow-hot-sun
+		),
+		green:(
+				spandex: $color-green-spandex
+		),
+);
```
Comment (kobizz) [inline core/app/assets/styles/tokens/maps/_colors-map.scss:16]: I don't think we need this depth

File: core/app/assets/styles/utilities/_align.scss [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// Utilities: Vertical align
+// - copied from "bootstrap 4.5
+
+.u-align-baseline    { vertical-align: baseline !important; } // Browser default
+.u-align-top         { vertical-align: top !important; }
+.u-align-middle      { vertical-align: middle !important; }
+.u-align-bottom      { vertical-align: bottom !important; }
+.u-align-text-bottom { vertical-align: text-bottom !important; }
+.u-align-text-top    { vertical-align: text-top !important; }
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_align.scss:9]: These are old-fashioned rules. We are not supposed to use them

File: core/app/assets/styles/utilities/_background.scss [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+	.u-bg-#{$color}{ background-color:var(--#{$color});}
+}
+
+@each $color, $value in $eps-tints {
+	.u-bg-gray-#{$color}{ background-color:var(--gray-#{$color});}
+}
+
+.u-bg-transparent {
+	background-color: transparent !important;
+}
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_background.scss:14]: What is it used for?

File: core/app/assets/styles/utilities/_borders.scss [hunk 9]
```diff
@@ -0,0 +1,74 @@
+// stylelint-disable property-blacklist, declaration-no-important
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_borders.scss:1]: I don't think we suppose to use all these rules. It's should be consumed as a variable wherever it's needed

File: core/app/assets/styles/utilities/_clearfix.scss [hunk 10]
```diff
@@ -0,0 +1,3 @@
+.u-clearfix {
+  @include clearfix();
+}
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_clearfix.scss:3]: Used?

File: core/app/assets/styles/utilities/_display.scss [hunk 11]
```diff
@@ -0,0 +1,17 @@
+// Utilities for common `display` values
+
+@each $value in $displays {
+	.u-d-#{$value} {
+		display: $value !important;
+	}
+}
+
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_display.scss:8]: Do we really need this method?

File: core/app/assets/styles/utilities/_embed.scss [hunk 12]
```diff
@@ -0,0 +1,39 @@
+// Credit: Nicolas Gallagher and SUIT CSS.
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_embed.scss:1]: What is it used for?

File: core/app/assets/styles/utilities/_flex.scss [hunk 13]
```diff
@@ -0,0 +1,139 @@
+// Flex variation
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_flex.scss:1]: I can't see the advantage of using these classes

File: core/app/assets/styles/utilities/_float.scss [hunk 14]
```diff
@@ -0,0 +1,13 @@
+// Utilities: float
```
Comment (kobizz) [inline core/app/assets/styles/utilities/_float.scss:1]: We should discuss all of these utility classes

**Review by kobizz — CHANGES_REQUESTED** [review]

File: .grunt-config/loaders/app-imports.js [hunk 15]
```diff
@@ -3,7 +3,7 @@ const dir = path.resolve( __dirname, '../../' );
 const fs = require('fs');
 
 module.exports = function() {
-	const importsFile = dir + '/core/app/assets/scss/app-imports.scss',
+	const importsFile = dir + '/core/app/assets/styles/app-imports.scss',
```
[the remark this answers is not in the document]
Comment (matipojo) [reply .grunt-config/loaders/app-imports.js:6]: We can change the convention - but for all modules.

File: core/app/assets/js/app.scss [hunk 16]
```diff
@@ -1,15 +1,83 @@
+@import "../styles/common";
+@import "../styles/base";
+@import "../styles/utilities";
+
+$app-background-color: tints(200);
+$app-sidebar-background-color: rgba(theme-colors(light), $opacity-5);
+$box-shadow-color: rgba(theme-colors(dark), $opacity-05);
+$app-sidebar-box-shadow: 0 2px 30px var(--box-shadow-color);
```
[the remark this answers is not in the document]
Comment (joshmarom) [reply core/app/assets/js/app.scss:8]: Box shadow tokens and functions are still WIP

File: core/app/assets/js/app.scss [hunk 17]
```diff
@@ -1,15 +1,83 @@
...
+.dialog-type-lightbox {
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
+
+.elementor-templates-modal .dialog-message {
+	height: 920px;
+}
+
```
[the remark this answers is not in the document]
Comment (joshmarom) [reply core/app/assets/js/app.scss:48]: This is temporary, just to get the view normalized, until further discussion.

**Pull request merged**